Src/Plugins/Security/SEC101_025.SendGridApiKeyValidator.cs

```diff
@@ -51,11 +51,14 @@ public static ValidationState IsValidDynamic(ref Fingerprint fingerprint, ref st
                                                       out Fingerprint fingerprint)
         {
             fingerprint = default;
-            string key = groups["key"];
+            if (!groups.TryGetValue("secret", out string secret))
```

Comment:
Shoudl this be "TryGetNonEmptyValue" ?

Reply:
in theory, the regex would only be caught if secret exists. so, IMO, we would not need it

---
In reply to: [605118483](https://github.com/microsoft/sarif-pattern-matcher/pull/349#discussion_r605118483) [](ancestors = 605118483)

Reply:
True, but regex may change in the future.  Excluding null/empty results improves our fault tolerance.  No huge performance hit so there's no reason to NOT do it, IMO

---
In reply to: [605119939](https://github.com/microsoft/sarif-pattern-matcher/pull/349#discussion_r605119939) [](ancestors = 605119939,605118483)